plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java

```diff
@@ -279,6 +280,28 @@ public Boolean getIsModelEnabled(String modelId) {
         return modelCache.getIsModelEnabled();
     }
 
+    /**
+     * Set a flag to show if model is in auto deploying status
+     *
+     * @param modelId        model id
+     * @param isModelAutoDeploying auto deploy flag
+     */
+    public synchronized void setIsAutoDeploying(String modelId, Boolean isModelAutoDeploying) {
+        log.debug("Setting the auto deploying flag for Model {}", modelId);
+        getExistingModelCache(modelId).setIsAutoDeploying(isModelAutoDeploying);
+    }
+
+    /**
+     * Check if model is in auto deploying.
+     *
+     * @param modelId model id
+     * @return true if model is auto deploying.
+     */
+    public boolean isAutoDeploying(String modelId) {
+        MLModelCache modelCache = modelCaches.get(modelId);
+        return modelCache != null && modelCache.getIsAutoDeploying() == true;
```

Comment:
```suggestion
        return modelCache != null && BooleanUtils.isTrue(modelCache.getIsAutoDeploying());
```

Reply:
Updated to "modelCache.getIsAutoDeploying() != null && BooleanUtils.isTrue(modelCache.getIsAutoDeploying())", otherwise it will throw an Null Pointer Exception.

Reply:
Updated.